dask/array/einsumfuncs.py

```diff
@@ -200,6 +200,7 @@ def _einsum_kernel(*operands, **kwargs):
 def einsum(*operands, **kwargs):
     dtype = kwargs.get('dtype')
     optimize = kwargs.get('optimize')
+    split_every = kwargs.pop('split_every', None)
     einsum_dtype = dtype
```

Comment:
Can we raise TypeError if there are unexpected keyword arguments?

Reply:
Any thoughts on this comment @crusaderky ?

Reply:
I think we shouldn't - blinly passing everything makes dask forward compatible with later versions of numpy. If you pass an unexpected parameter, you'll still get a proper error message at graph resolution time:
```
>>> a = da.random.random(10, chunks=5)
>>> b = da.einsum('i', a, foo=False)
dask.array<einsum_kernel, shape=(10,), dtype=float64, chunksize=(5,)>
>>> b.compute()
TypeError: 'foo' is an invalid keyword for einsum
```

Reply:
It's a good point.  However in general the project has so far decided to try to err early rather than at compute time when it's possible to do so.  This ends up being important because otherwise it becomes difficult to track where the error occurred for large and complex workflows (which are common).  I recommend that we only accept those keywords that are explicitly supported today and change them if the Numpy API ever changes (which seems to happen slowly anyway).

Reply:
Alternatively perhaps you could call np.einsum with an empty array of the correct dimension with the leftover kwargs to raise whatever error would normally be raised?  This would achieve your goal of being future-compatible while also providing informative errors during graph construction.

Reply:
I believe that leftover kwargs is the only issue currently holding up this PR.